### ISSUE TITLE
Amplify push doesn't work

### ISSUE THREAD

**Issue by YikSanChan** [aws-amplify/amplify-cli#553]
**Describe the bug**
In my schema.graphql, I added `@searchable`. However, running `amplify push` doesn't actually update cloud resource even though it says so.

**To Reproduce**
1. Update my current schema.graphql to add `@searchable` after `@model`.
2. `amplify push`.
```
| Category | Resource name      | Operation | Provider plugin   |
| -------- | ------------------ | --------- | ----------------- |
| Api      | inventoryappmobile | Update    |                   |
| Auth     | cognitoc7f37f50    | No Change | awscloudformation |
| Storage  | s307ab5088         | No Change | awscloudformation |
? Are you sure you want to continue? Yes
✔ All resources are updated in the cloud
```
3. `amplify codegen statements --nodownload` because I want to use GraphQL query such as `searchXX` in my code. (See [here](https://aws-amplify.github.io/docs/cli/codegen#amplify-codegen-statements-))
```
✔ Generated GraphQL operations successfully and saved at src/graphql
```

**Expected behavior**
Cloud resource should be updated and GraphQL statements should be updated to include searchXX. However, after above steps, I run `amplify status` and it still shows:
```
| Category | Resource name      | Operation | Provider plugin   |
| -------- | ------------------ | --------- | ----------------- |
| Api      | inventoryappmobile | Update    |                   |
| Auth     | cognitoc7f37f50    | No Change | awscloudformation |
| Storage  | s307ab5088         | No Change | awscloudformation |
```
I believe cloud resources are not updated and same to GraphQL statements.

**Additional context**
amplify version: 0.1.36
OS: MacOS

**Comment by yuth** [comment]
How did you add the API to your project. Was it using transformer (`amplify add api`) or using using API_ID (`amplify add codegen --apiId <API_ID>`)

**Comment by YikSanChan** [comment]
First `amplify add api` to add GraphQL backend, then `amplify add codegen --apiId <API_ID>` to add codegen.

**Comment by yuth** [comment]
You don't need to add codegen if you're have added the `api ` using amplify. Amplify supports only one of the commands in a project. (If you have added a GraphQL API using `amplify add api` then you won't be allowed to run `amplify codegen add --apiId <API_ID>`. 

From the look of status I see that your project is retaining only the `codegen add` and it can't update api to in the cloud.

I am curious how you managed to run both `amplify api add` and `amplify codegen add`.

**Comment by YikSanChan** [comment]
The only fact I know about is, I did both, and after I did the second, a `.graphqlconfig.yml` is generated:

```
projects:
  <project_name>:
    schemaPath: src/graphql/schema.json
    includes:
      - src/graphql/**/*.js
    excludes:
      - ./amplify/**
    extensions:
      amplify:
        graphQLApiId: <api_id>
        codeGenTarget: javascript
        generatedFileName: ''
        docsFilePath: src/graphql
        region: us-west-2
      endpoints:
        prod: >-
          https://tiimlhlswngxjj3c3siwbqw3ji.appsync-api.us-west-2.amazonaws.com/graphql
```

`amplify/#current-cloud-backend/amplify-meta.json` and `amplify/backend/amplify-meta.json` are updated.

`src/graphql/schema.json`, `src/graphql/subscriptions.js`, `src/graphql/queries.js` and `src/graphql/mutations.js` are generated.

Any help? Thanks

**Comment by yuth** [comment]
It might be possible that you declined codegen when push command was run. Let me try to see if that's the case

**Comment by YikSanChan** [comment]
Anything I can do from my end?

**Comment by yuth** [comment]
If you don't mind loosing your change that you did last you could try the to recover. Make sure you __backup__ your folder before trying this as things might not work as expected

Copy all the content from your `<proj-root>/amplify/#current-cloud-backend` to `<proj-root>/amplify/backend`

Running `amplify ls` you see `Provider plugin` for the API row

**Comment by YikSanChan** [comment]
@yuth Copy from `<proj-root>/amplify/#current-cloud-backend` to `<proj-root>/amplify/backend` doesn't work because they are exactly the same. `amplify ls` always show below.

```
| Category | Resource name      | Operation | Provider plugin   |
| -------- | ------------------ | --------- | ----------------- |
| Api      | inventoryappmobile | Update    |                   |
| Auth     | cognitoc7f37f50    | No Change | awscloudformation |
| Storage  | s307ab5088         | No Change | awscloudformation |

GraphQL endpoint: https://tiimlhlswngxjj3c3siwbqw3ji.appsync-api.us-west-2.amazonaws.com/graphql
```

**Comment by YikSanChan** [comment]
Also, I find the same issue #243 but after I try `amplify api gql-compile`, still nothing happens. Seems @mkaschke and @jtaylor1989 are faced with the same issues. Really look forward to solving the problem

**Comment by YikSanChan** [comment]
Resolved in #565
